gson/src/main/java/com/google/gson/internal/LinkedTreeMap.java

```diff
@@ -459,7 +459,7 @@ static final class Node<K, V> implements Entry<K, V> {
     }
 
     /** Create a regular entry */
-    Node(Node<K, V> parent, K key, Node<K, V> next, Node<K, V> prev) {
+    Node(Node<K, V> parent, K key, @Weak Node<K, V> next, @Weak Node<K, V> prev) {
```

Comment:
In members declaration, `@weak` is used on parent, next and prev.
Here it's used only on next and prev.
Maybe it's good but just wondering, is it intended?